File: matlab/examples/gpu_asteroid_validation.m
function gpu_asteroid_validation()
% GPU_ASTEROID_VALIDATION  Summary of Keplerian-vs-Horizons error samples.
%
% Reads examples/gpu-asteroid-swarm/data/validation_sample.json produced by
% scripts/validate_against_horizons.py and prints a table.

  root = fileparts(fileparts(fileparts(mfilename('fullpath'))));
  jsonPath = fullfile(root, 'examples', 'gpu-asteroid-swarm', 'data', 'validation_sample.json');
  if ~exist(jsonPath, 'file')
    error('Run scripts/validate_against_horizons.py first (missing %s)', jsonPath);
  end
  payload = jsondecode(fileread(jsonPath));
  fprintf('Validation date: %s  (JD %.1f)\n', payload.date, payload.jd);
  fprintf('%-26s %9s %12s\n', 'Object', 'dT [yr]', '|dr| [AU]');
  fprintf('%s\n', repmat('-', 1, 50));
  for k = 1:numel(payload.samples)
    s = payload.samples(k);
    fprintf('%-26s %+9.2f %12.4e\n', s.name, s.dt_years, s.err_au);
  end
end

File: python/examples/gpu_asteroid_validation.py
#!/usr/bin/env python3
"""Short-horizon validation: pure-Kepler vs JPL Horizons on named asteroids.

This is a thin wrapper around scripts/validate_against_horizons.py that
reads the resulting validation_sample.json and prints a summary table.
Useful for quick sanity-checking after regenerating the data.
"""
from __future__ import annotations

import json
from pathlib import Path

DEFAULT = (Path(__file__).resolve().parents[2]
           / "examples/gpu-asteroid-swarm/data/validation_sample.json")


def main(path: Path = DEFAULT) -> None:
    if not path.exists():
        print(f"{path} not found — run scripts/validate_against_horizons.py first")
        return
    payload = json.loads(path.read_text())
    print(f"Validation date: {payload['date']}  (JD {payload['jd']:.1f})")
    print(f"{'Object':26s} {'ΔT [yr]':>9s} {'|Δr| [AU]':>12s}")
    print("-" * 50)
    for s in payload["samples"]:
        print(f"{s['name']:26s} {s['dt_years']:+9.2f} {s['err_au']:12.4e}")


if __name__ == "__main__":
    main()
